system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -81,7 +81,7 @@ this.TopStoriesFeed = class TopStoriesFeed {
               "image": this._normalizeUrl(s.image_src),
               "referrer": this.stories_referrer,
               "url": s.url,
-              "eTLD": this._addETLD(s.url),
+              "eTLD": getETLD(s.url),
```

Comment:
Your previous PR #2987 added the hostname field in `TopStoriesFeed` but actually regressed #3031 because the eTLD no longer existed by the time shortURL was called.

https://github.com/piatra/activity-stream/commit/d782ac9e6ea4e720b689b440300a2c38d9f8a4a1#diff-506774124923d486589dbeda753ebe38L51

This line is no longer necessary as `shortURL` will get the `eTLD` as necessary.